test/lit/passes/gto_and_cfp_in_O.wast

```diff
@@ -1,18 +1,38 @@
 ;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
 
-;; RUN: foreach %s %t wasm-opt -O -all --nominal -S -o - | filecheck %s
+;; RUN: foreach %s %t wasm-opt -O -all --nominal                -S -o - | filecheck %s --check-prefix OPENED
+;; RUN: foreach %s %t wasm-opt -O -all --nominal --closed-world -S -o - | filecheck %s --check-prefix CLOSED
 
 ;; Test that -O, with nominal typing + GC enabled, will run global type
-;; optimization in conjunction with constant field propagation etc.
+;; optimization in conjunction with constant field propagation etc. For GTO,
+;; we also need the closed world flag, as otherwise we will not modify the shape
+;; of struct types.
 
 (module
+  ;; OPENED:      (type $struct (struct_subtype (field (mut (ref $none_=>_none))) (field (mut i32)) data))
   (type $struct (struct_subtype (field (mut funcref)) (field (mut i32)) data))
 
+  ;; OPENED:      (type $none_=>_none (func_subtype func))
+
+  ;; OPENED:      (type $none_=>_i32 (func_subtype (result i32) func))
+
+  ;; OPENED:      (global $glob (ref $struct) (struct.new $struct
+  ;; OPENED-NEXT:  (ref.func $by-ref)
+  ;; OPENED-NEXT:  (i32.const 100)
+  ;; OPENED-NEXT: ))
   (global $glob (ref $struct) (struct.new $struct
     (ref.func $by-ref)
     (i32.const 100)
   ))
 
+  ;; OPENED:      (export "main" (func $main))
+
+  ;; OPENED:      (func $by-ref (type $none_=>_none) (; has Stack IR ;)
+  ;; OPENED-NEXT:  (struct.set $struct 1
+  ;; OPENED-NEXT:   (global.get $glob)
+  ;; OPENED-NEXT:   (i32.const 200)
+  ;; OPENED-NEXT:  )
+  ;; OPENED-NEXT: )
```

Comment:
I wonder why `CLOSED` check lines are generated with this function?

Reply:
Do you mean why there are no `CLOSED` check lines for this function? It's because it is optimized out with the closed world flag and does not exist in that output.